src/ccl_power.c

```diff
@@ -1281,21 +1115,23 @@ TASK: compute the nonlinear power spectrum at a given redshift
 
 double ccl_nonlin_matter_power(ccl_cosmology * cosmo, double k, double a, int *status)
 {
-  switch(cosmo->config.matter_power_spectrum_method) {
-
-    //If the matter PS specified was linear, then do the linear compuation
-  case ccl_linear:
-    return ccl_linear_matter_power(cosmo,k,a,status);
+  
+  //If the matter PS specified was linear, then do the linear compuation
+  if(cosmo->config.matter_power_spectrum_method==ccl_linear) {
     
-  case ccl_halofit:
+    return ccl_linear_matter_power(cosmo,k,a,status);
+
+  }
+  else if (cosmo->config.matter_power_spectrum_method==ccl_halofit || cosmo->config.matter_power_spectrum_method==ccl_baryons){
```

Comment:
We should think about what's gonna happen if we want to use the emulator AND baryons. Currently baryons means halofit+baryons... We could make ccl_baryons a separate flag (not related to a PS method). What do you think?